src/message.rs

```diff
@@ -1369,6 +1369,58 @@ pub fn get_deaddrop_msg_cnt(context: &Context) -> usize {
     }
 }
 
+pub fn estimate_deletion_cnt(
+    context: &Context,
+    from_server: bool,
+    seconds: i64,
+) -> Result<usize, Error> {
+    let self_chat_id = chat::lookup_by_contact_id(context, DC_CONTACT_ID_SELF)
+        .unwrap_or_default()
+        .0;
+    let threshold_timestamp = time() - seconds;
+
+    let cnt: isize;
+    if from_server {
+        cnt = context.sql.query_row(
+            "SELECT COUNT(*)
+             FROM msgs m
+             WHERE m.id > ?
+               AND (state = ? OR state >= ?)
```

Comment:
What is the puprose of these state checks? I am deleting messages regardless of whether they are seen or not.

Reply:
you're right, we should keep things simple currently, i will remove them but this may need some additional discussions - mainly for deletion on the device in combination with short life-times.